From 3106628c42b024a87ab46045f21cb2f3edaf4557 Mon Sep 17 00:00:00 2001
From: Ye Bin <yebin10@huawei.com>
Date: Wed, 24 Feb 2021 04:32:48 -0500
Subject: [PATCH] arm32: kaslr: pop visibility when compile decompress boot
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 3106628c42b024a87ab46045f21cb2f3edaf4557

 code as we need relocate BSS by GOT.

hulk inclusion
commit 6337511516862e5a4d2d5a96481510e4a7a12b1b
category: feature
feature: ARM kaslr support
bugzilla: 47952
CVE: NA

-------------------------------------------------

If we not hide the GOT, when insert module which reference global
variable we got error "Unknown symbol_GLOBAL_OFFSET_TABLE_ (err 0)".

Signed-off-by: Ye Bin <yebin10@huawei.com>
Reviewed-by: Jing Xiangfeng <jingxiangfeng@huawei.com>
Signed-off-by: zhangyi (F) <yi.zhang@huawei.com>
Signed-off-by: Cui GaoSheng <cuigaosheng1@huawei.com>
Reviewed-by: Xiu Jianfeng <xiujianfeng@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm/boot/compressed/decompress.c | 4 ++++
 arch/arm/boot/compressed/misc.c       | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/arch/arm/boot/compressed/decompress.c b/arch/arm/boot/compressed/decompress.c
index aa075d8372ea..38a5dd847e03 100644
--- a/arch/arm/boot/compressed/decompress.c
+++ b/arch/arm/boot/compressed/decompress.c
@@ -1,6 +1,10 @@
 // SPDX-License-Identifier: GPL-2.0
 #define _LINUX_STRING_H_
 
+#ifdef CONFIG_RANDOMIZE_BASE
+#pragma GCC visibility pop
+#endif
+
 #include <linux/compiler.h>	/* for inline */
 #include <linux/types.h>	/* for size_t */
 #include <linux/stddef.h>	/* for NULL */
diff --git a/arch/arm/boot/compressed/misc.c b/arch/arm/boot/compressed/misc.c
index e1e9a5dde853..974a08df7c7a 100644
--- a/arch/arm/boot/compressed/misc.c
+++ b/arch/arm/boot/compressed/misc.c
@@ -16,6 +16,9 @@
  *  which should point to addresses in RAM and cleared to 0 on start.
  *  This allows for a much quicker boot time.
  */
+#ifdef CONFIG_RANDOMIZE_BASE
+#pragma GCC visibility pop
+#endif
 
 unsigned int __machine_arch_type;
 
-- 
2.26.2

